e2e/stack.go

```diff
@@ -266,3 +283,51 @@ func (s *Stack) runMonomer(ctx context.Context, env *environment.Env, genesisTim
 	}
 	return nil
 }
+
+type mockAccountRetriever struct {
+	ReturnAccNum, ReturnAccSeq uint64
+}
+
+func (mar mockAccountRetriever) GetAccount(_ client.Context, _ sdktypes.AccAddress) (client.Account, error) { //nolint:gocritic // hugeParam
+	return mockAccount{}, nil
+}
+
+func (mar mockAccountRetriever) GetAccountWithHeight(
+	_ client.Context, //nolint:gocritic // hugeParam
+	_ sdktypes.AccAddress,
+) (client.Account, int64, error) {
+	return mockAccount{}, 0, nil
+}
+
+func (mar mockAccountRetriever) EnsureExists(_ client.Context, _ sdktypes.AccAddress) error { //nolint:gocritic // hugeParam
+	return nil
+}
+
+func (mar mockAccountRetriever) GetAccountNumberSequence(
+	_ client.Context, //nolint:gocritic // hugeParam
+	_ sdktypes.AccAddress,
+) (uint64, uint64, error) {
+	return mar.ReturnAccNum, mar.ReturnAccSeq, nil
+}
+
+type mockAccount struct {
+	address   sdktypes.AccAddress
+	seqNumber uint64
+}
+
+func (mar mockAccount) GetAddress() sdktypes.AccAddress {
+	return mar.address
+}
+
+func (mar mockAccount) GetPubKey() cryptotypes.PubKey {
+	return nil
+}
+
+func (mar mockAccount) GetAccountNumber() uint64 {
+	return 1
+}
+
+func (mar mockAccount) GetSequence() uint64 {
+	mar.seqNumber++
```

Comment:
This is probably something that we don't need to worry about since it's just a mock account but I wanted to leave a comment just in case. If monomer errors out after a tx is signed but before it's sent will the sequence number get out of sync?

Reply:
Cosmos seqNumbers on transactions don't have to be in-order (`1, 2, 3, ...`), they just have to be increasing (`1, 2, 7, 8, ...`). I think this implementation will satisfy that.